lib/SIL/Verifier/SILVerifier.cpp

```diff
@@ -662,6 +662,12 @@ class SILVerifier : public SILVerifierBase<SILVerifier> {
 
   // Used for dominance checking within a basic block.
   llvm::DenseMap<const SILInstruction *, unsigned> InstNumbers;
+  
+  // Map of all the get_async_continuation[_addr] instructions in async
+  // functions, for checking suspendable contexts.
+  llvm::DenseMap<SILFunction *, SmallVector<SingleValueInstruction *, 2>>
+    AsyncContinuationSources;
```

Comment:
I think you can just do this analysis in `checkFlowSensitiveRules`, and then you won't need to kindof reinvent it.